scripts/tools/IndexMissingRecords.php

```diff
@@ -0,0 +1,205 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoAdvancedSearch\scripts\tools;
+
+use core_kernel_classes_Resource;
+use oat\generis\model\kernel\persistence\smoothsql\search\ComplexSearchService;
+use oat\generis\model\OntologyAwareTrait;
+use oat\oatbox\extension\script\ScriptAction;
+use oat\oatbox\reporting\Report;
+use oat\search\base\ResultSetInterface;
+use oat\tao\elasticsearch\ElasticSearch;
+use oat\tao\elasticsearch\IndexerInterface;
+use oat\tao\elasticsearch\Query;
+use oat\tao\model\search\SearchProxy;
+use oat\tao\model\TaoOntology;
+use oat\taoAdvancedSearch\model\Resource\Service\SyncResourceResultIndexer;
+use Zend\ServiceManager\ServiceLocatorAwareInterface;
+use Zend\ServiceManager\ServiceLocatorAwareTrait;
+
+/**
+ * php index.php 'oat\taoAdvancedSearch\scripts\tools\IndexMissingRecords' -h
+ */
+class IndexMissingRecords extends ScriptAction implements ServiceLocatorAwareInterface
+{
+    use ServiceLocatorAwareTrait;
+    use OntologyAwareTrait;
+
+    protected function provideUsage(): array
+    {
+        return [
+            'prefix' => 'h',
+            'longPrefix' => 'help',
+            'description' => 'Type this option to see the parameters.'
+        ];
+    }
+
+    protected function provideOptions(): array
+    {
+        return [
+            'class' => [
+                'prefix' => 'c',
+                'longPrefix' => 'class',
+                'flag' => false,
+                'description' => 'Resource class URI',
+                'defaultValue' => TaoOntology::CLASS_URI_ITEM
+            ],
+            'reindex' => [
+                'prefix' => 'r',
+                'longPrefix' => 'reindex',
+                'flag' => false,
+                'description' => 'Force reindex missing resources',
+                'defaultValue' => false
+            ],
+            'offset' => [
+                'prefix' => 'o',
+                'longPrefix' => 'offset',
+                'flag' => false,
+                'description' => 'offset of search results',
+                'defaultValue' => 0
+            ],
+            'limit' => [
+                'prefix' => 'l',
+                'longPrefix' => 'limit',
+                'flag' => false,
+                'description' => 'limit of search results',
+                'defaultValue' => 0
+            ]
+        ];
+    }
+
+    protected function provideDescription(): string
+    {
+        return 'Show indexation summary';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    protected function run(): Report
+    {
+        $class = $this->getOption('class');
+        $offset = (int)$this->getOption('offset');
+        $limit = (int)$this->getOption('limit');
```

Comment:
You can cast them in `provideOptions` method ([see more](https://github.com/oat-sa/generis/blob/master/common/oatbox/extension/script/OptionContainer.php#L172)).
Use:
```php
'offset' => [
    'cast' => 'int', // or 'cast' => 'integer',
],
'limit' => [
    'cast' => 'int', // or 'cast' => 'integer',
],
```